### DISCUSSION TITLE
gtfToGenePred error

### DISCUSSION THREAD

**Discussion by TheCatalyticMicroChip** [ConesaLab/SQANTI3#343] [Q&A] [answered]
I am trying to run python sqanti3_qc.py in a linux HPC. I encounter the following error: 
`(ENV) [user@gra736 SQANTI3-5.2.2]$ python sqanti3_qc.py --gtf B2.collapsed.sorted.gff gencode.mouse_GRCm39.sorted.gtf ~mouse_GRCm39.fasta --fl_count B2.collapsed.abundance.txt
sqanti3_qc.py:75: DeprecationWarning: Use shutil.which instead of find_executable
  if distutils.spawn.find_executable(GTF2GENEPRED_PROG) is None:
sqanti3_qc.py:78: DeprecationWarning: Use shutil.which instead of find_executable
  if distutils.spawn.find_executable(GFFREAD_PROG) is None:
Cannot find executable gffread. Abort!
`.
When I manually changed the script to:
 `import shutil
GTF2GENEPRED_PROG = os.path.join(utilitiesPath,"gtfToGenePred")
GFFREAD_PROG = shutil.which("gffread")
if not GTF2GENEPRED_PROG:
    print("Cannot find executable {0}. Abort!".format(GTF2GENEPRED_PROG), file=sys.stderr)
    sys.exit(-1)
if not GFFREAD_PROG:
    print("Cannot find executable {0}. Abort!".format(GFFREAD_PROG), file=sys.stderr)
    sys.exit(-1)` 
I get "Cannot find executable None. Abort!"

**Comment by handesome** [comment]
Hi!did you solve the problem?I encounter the errorsays` FileNotFoundError:` [Errno 2] No such file or directory: /data/work/temp2/sqan/refAnnotation_./output.genePred'`,could you give some advice?

**Reply by TheCatalyticMicroChip** [reply]
Unfortunately, not.

**Reply by handesome** [reply]
so bad，wanna cry

**Comment by lkondratova** [comment, marked as answer]
Hi @TheCatalyticMicroChip,
It's difficult to say what the issue could be. A few things I would suggest are to make sure that the SQANTI3.env is activated, that the files are executable (you can run chmod -R 777 sqanti_folder to ensure this), and if you are not running SQANTI3 from its original folder, make sure you set the symbolic link to the SQANTI3 folder. Additionally, ensure that the SQANTI3 folder was downloaded properly and that /SQANTI3/utilities/gtfToGenePred exists.
